trie/utils/binaries.py

```diff
@@ -0,0 +1,66 @@
+from trie.constants import (
+    TWO_BITS,
+    PREFIX_00,
+    PREFIX_100000,
+)
+
+
+def safe_ord(value):
+    if isinstance(value, int):
+        return value
+    else:
+        return ord(value)
+
+
+def decode_from_bin(input_bin):
+    """
+    0100000101010111010000110100100101001001 -> ASCII
+    """
+    output = bytearray(len(input_bin) // 8)
+    for i in range(0, len(input_bin), 8):
+        byte_value = 0
+        for bit_value in input_bin[i:i+8]:
+            byte_value = byte_value * 2 + bit_value
+        output[i//8] = byte_value
+    return bytes(output)
+
+
+def encode_to_bin(value):
```

Comment:
Ended up also taking a stab at an alternate implementation of this.

```python
EXP = tuple(reversed(tuple(2**i for i in range(8))))


@apply_to_return_value(bytes)
def alt_encode_to_bin(value):
    for char in value:
        for exp in EXP:
            if char & exp:
                yield True
            else:
                yield False
```

Benchmarking suggests it's 5-20% faster than the current implementation and I think it's a bit easier to read as well.